### PR TITLE
Rust: Add Support for Path and Steps

### PR DESCRIPTION
**Pull request by droogmic**
Small increment to support way 40297361

### CODE REVIEW

File: rust/osm2lanes/src/transform/tags_to_lanes/mod.rs [hunk 1]
```diff
@@ -449,9 +449,6 @@ impl RoadBuilder {
                 }
                 _ => todo!(),
             };
-            let forward_edge = dbg!(forward_edge);
-            let middle_separator = dbg!(middle_separator);
-            let backward_edge = dbg!(backward_edge);
```
Comment (droogmic) [inline rust/osm2lanes/src/transform/tags_to_lanes/mod.rs:454]: I need to find a way to stop this being committed in the future.

File: rust/osm2lanes/src/transform/lanes_to_tags/mod.rs [hunk 2]
```diff
@@ -56,6 +56,12 @@ impl Lane {
 pub fn lanes_to_tags(lanes: &[Lane], locale: &Locale, config: &Config) -> TagsResult {
     let mut tags = Tags::default();
     let mut oneway = false;
+
+    if !lanes.iter().any(|lane| lane.is_motor() || lane.is_bus()) {
+        tags.checked_insert("highway", "path")?;
```
Comment (dabreegster) [inline rust/osm2lanes/src/transform/lanes_to_tags/mod.rs:61]: What if this method took `&Road` and we just used `Highway` here? This seems super brittle; what if it was `steps` or some other kind of footway?

File: rust/osm2lanes/src/transform/tags_to_lanes/mod.rs (hunk 1 above)
Comment (dabreegster) [reply rust/osm2lanes/src/transform/tags_to_lanes/mod.rs:454]: Ah, I noticed in code review but didn't comment, my mistake. If it was a big enough problem, we could have a presubmit check grep for it or something

File: rust/osm2lanes/src/transform/lanes_to_tags/mod.rs (hunk 2 above)
Comment (droogmic) [reply rust/osm2lanes/src/transform/lanes_to_tags/mod.rs:61]: yeah, this will need to change, will refactor as part of a future PR :)

**Review by dabreegster — APPROVED** [review]

File: rust/osm2lanes/src/transform/tags_to_lanes/mod.rs (hunk 1 above)
Comment (BudgieInWA) [reply rust/osm2lanes/src/transform/tags_to_lanes/mod.rs:454]: That sounds like the way to go! I noticed [Jon Blow](https://www.youtube.com/user/jblow888) using `// @nocheckin` (and related) in his setup and thought it was a useful idea.

**Pull request merged**